### ISSUE TITLE
Update videos on homepage

### ISSUE THREAD

**Issue by kosmikus** [haskell-infra/www.haskell.org#288]
I'll be honest: My primary motivation is to make "The Haskell Unfolder" (https://www.youtube.com/playlist?list=PLD8gywOEY4HaG5VSrKVnHxCptlJv2GAn7) more visible.

However, I generally think that the videos currently on the homepage have been there for a long time, and if we want to make Haskell look like an actively used language with a vibrant community, we should aim to choose recent videos even if the old ones are without doubt good. It just makes a better impression to a visitor of the page if the videos are not all a decade old.

Here are a few suggestions for good videos and playlists that I could quickly come up with. My perspective is limited, I'm almost certainly forgetting many good options. I've deliberately included playlists, and videos targeted at different levels / backgrounds. My suggestion would be to generally update more often, or possibly even have a larger selection and randomly pick a few on every visit (the SPJ video is currently there already; I think it's good, and I didn't find a more recent and equally relevant SPJ video, so I'd keep it):

Functional Programming & Haskell - Computerphile (John Hughes)
https://www.youtube.com/watch?v=LnX3B9oaKzw

Escape from the Ivory Tower / The Haskell Journey (Simon Peyton Jones)
https://www.youtube.com/watch?v=re96UgMk6GQ

Past and Present of Haskell - Interview with Simon Peyton Jones
https://www.youtube.com/watch?v=4RuLzL_q0zs

The Haskell Unfolder (Andres Löh and Edsko de Vries)
https://www.youtube.com/playlist?list=PLD8gywOEY4HaG5VSrKVnHxCptlJv2GAn7

Functional Programming in Haskell (Graham Hutton)
https://www.youtube.com/playlist?list=PLF1Z-APd9zK7usPMx3LGMZEHrECUGodd3

Advanced Functional Programming in Haskell (Graham Hutton)
https://www.youtube.com/playlist?list=PLF1Z-APd9zK5uFc8FKr_di9bfsYv8-lbc

Laziness in Haskell (Alexis King)
https://www.youtube.com/playlist?list=PLyzwHTVJlRc8620PjqbM0x435-6-Gi1Gu

Delimited Continuations Demystified (Alexis King)
https://www.youtube.com/watch?v=DRFsodbxHQo&list=PLOvRW_utVPVmzDGGOJ2amgVBK168Vemke&index=3

What is a Monad? - Computerphile (Graham Hutton)
https://www.youtube.com/watch?v=t1e8gqXLbsU

Monad transformers are good, actually (Gabriella Gonzalez)
https://www.youtube.com/watch?v=w9ExsWcoXPs&list=PLOvRW_utVPVmzDGGOJ2amgVBK168Vemke&index=4

Make it Purple: An Introduction to Type Level Programming (Rebecca Skinner)
https://www.youtube.com/watch?v=fQggLFtdTTo

**Comment by tomjaguarpaw** [comment]
Yes, this has been a thorn in our side for a long time. We've discussed refreshing the videos many times, but no one has ever got round to doing it. I think we should focus on videos more suited to people unfamiliar with Haskell.

Thank you for putting together the list. How does this selection strike you? And could we add to it one specific video from The Haskell Unfoldr that is aimed mostly for people less-familiar with Haskell?

Escape from the Ivory Tower / The Haskell Journey (Simon Peyton Jones)
https://www.youtube.com/watch?v=re96UgMk6GQ

Functional Programming & Haskell - Computerphile (John Hughes)
https://www.youtube.com/watch?v=LnX3B9oaKzw

Past and Present of Haskell - Interview with Simon Peyton Jones
https://www.youtube.com/watch?v=4RuLzL_q0zs

Functional Programming in Haskell (Graham Hutton) (I'm concerned this one may go in and out of availability as Graham teaches his real-world course)
https://www.youtube.com/playlist?list=PLF1Z-APd9zK7usPMx3LGMZEHrECUGodd3

What is a Monad? - Computerphile (Graham Hutton)
https://www.youtube.com/watch?v=t1e8gqXLbsU

**Comment by kosmikus** [comment]
Thanks for the prompt reply. I am not sure I agree that videos on the homepage should be *exclusively* aimed at beginners. Sometimes, you get a better impression of a language via something that's not a classic tutorial or overview talk. I did try to select talks that are appealing to a wide audience (within Haskell) and cover topics that are interesting to many people (monads, laziness, type-level programming).

The Unfolder has episodes targeted at many different levels. There are quite a few "beginner" episodes, although most require at least some basic understanding of Haskell. I guess one that is understandable with relatively few prerequisites is the one on parametricity https://www.youtube.com/watch?v=mFUQYXqaODw&list=PLD8gywOEY4HaG5VSrKVnHxCptlJv2GAn7&index=12.

Other episodes I'd consider relatively beginner-friendly are:

- "learning by testing" https://www.youtube.com/watch?v=mFUQYXqaODw&list=PLD8gywOEY4HaG5VSrKVnHxCptlJv2GAn7&index=12
- "unfoldr" https://www.youtube.com/watch?v=S_HSt6jEtWM&list=PLD8gywOEY4HaG5VSrKVnHxCptlJv2GAn7&index=1
- "a new perspective on foldl'" https://www.youtube.com/watch?v=pQThkisvsVs&list=PLD8gywOEY4HaG5VSrKVnHxCptlJv2GAn7&index=19
- "Dijkstra's shortest paths" https://www.youtube.com/watch?v=WHVMjMIJnB4&list=PLD8gywOEY4HaG5VSrKVnHxCptlJv2GAn7&index=20

**Comment by kosmikus** [comment]
Another argument in favour of not exclusively having beginner-oriented videos: these days, not every newcomer to Haskell is a beginner in functional programming. People are also coming to Haskell via Purescript, Scala, OCaml, F#, Rust(!), ..., and may very well have seen FP concepts and/or relatively strong type systems before.

**Comment by tomjaguarpaw** [comment]
I didn't mention beginners, rather people unfamiliar with Haskell, because people who are already familiar with Haskell will likely have others sources for the Haskell content they like. Content for people unfamiliar with the language may well be at an advanced level, but it shouldn't be discouraging for people who have come onto our homepage because they learned about Haskell five minutes ago.

So my next suggestion is this selection of six to replace the six currently on the front page. What do you think? I'd like to invite comments from the Haskell.org committee too.

Escape from the Ivory Tower / The Haskell Journey (Simon Peyton Jones)
https://www.youtube.com/watch?v=re96UgMk6GQ

Functional Programming & Haskell - Computerphile (John Hughes)
https://www.youtube.com/watch?v=LnX3B9oaKzw

Past and Present of Haskell - Interview with Simon Peyton Jones
https://www.youtube.com/watch?v=4RuLzL_q0zs

Functional Programming in Haskell (Graham Hutton) (I'm concerned this one may go in and out of availability as Graham teaches his real-world course)
https://www.youtube.com/playlist?list=PLF1Z-APd9zK7usPMx3LGMZEHrECUGodd3

What is a Monad? - Computerphile (Graham Hutton)
https://www.youtube.com/watch?v=t1e8gqXLbsU

"Dijkstra's shortest paths" (I like the suggestion of "Dijkstra's shortest paths", because some people may be wondering whether this "functional language" thing can implement "imperative" algorithms.)
https://www.youtube.com/watch?v=WHVMjMIJnB4&list=PLD8gywOEY4HaG5VSrKVnHxCptlJv2GAn7&index=20

Also cc'ing @goldfirere, because he's expressed an interest in this video issue before.

**Comment by kosmikus** [comment]
FWIW, I'm fine with your proposal. I still think a larger selection of videos would be better than to basically decide now what will be there for the next 10 years, but this selection works for me.

As for Graham's playlist: I think the points in time where really *no* video is available are minor, because he's usually just removing the old ones a day or so before the first videos of the new iterations become available. But the playlist URL itself might change. I don't know that. Perhaps we should ask Graham about it.

Linking to Richard's Tweag playlist is also an option, of course. Lots of good stuff there (but again, most of it is not exactly newcomer-friendly).

Also, I think the SkillsMatter recovery effort currently underway is likely to unearth a few more good video candidates.

**Comment by tomjaguarpaw** [comment]
> I'm fine with your proposal

OK, great.

> I still think a larger selection of videos would be better than to basically decide now what will be there for the next 10 years

The turnaround time will be quicker than 10 years if active contributors like you prompt us to change them! And besides, I don't think we should let uncertainty about the choice stop us from refreshing the selection _right now_. If we try to devise the "correct" policy then we're likely to languish indefinitely with no change of videos and no policy.

> As for Graham's playlist: I think the points in time where really no video is available are minor, because he's usually just removing the old ones a day or so before the first videos of the new iterations become available

OK, fair enough. We can deal with any problems if and when they arise.

> Linking to Richard's Tweag playlist is also an option, of course. Lots of good stuff there (but again, most of it is not exactly newcomer-friendly).
>
>Also, I think the SkillsMatter recovery effort currently underway is likely to unearth a few more good video candidates.

Both great ideas. Thanks for the suggestions! Let's go with the six suggested above for now, and then once we've got the ball rolling it will be easier to tweak things as time goes on.

Feel free to make a PR. Otherwise I'll do it when I get round to it (or some Haskell.org committee member might take it up).
